core/consensus/babe/primitives/src/lib.rs

```diff
@@ -30,22 +30,27 @@ use substrate_client::decl_runtime_apis;
 pub use digest::{BabePreDigest, CompatibleDigestItem};
 pub use digest::{BABE_VRF_PREFIX, RawBabePreDigest};
 
-mod app {
+mod app_sr25519 {
 	use app_crypto::{app_crypto, key_types::BABE, sr25519};
 	app_crypto!(sr25519, BABE);
 }
 
-/// A Babe authority keypair. Necessarily equivalent to the schnorrkel public key used in
-/// the main Babe module. If that ever changes, then this must, too.
-#[cfg(feature = "std")]
-pub type AuthorityPair = app::Pair;
+/// Sr25519 crypto for BABE
+pub mod sr25519 {
```

Comment:
What is the reason for this change? Babe only works with `sr25519`?

Reply:
I suppose to make it easier to mock. BABE could work with other kinds of crypto, but it's hard to reason about the necessary generalizations we'd have to make.

Reply:
@rphmeier Yup.  This way, I can use the existing `UintAuthorityId` infrastructure.

Reply:
I still don't understand why this is needed. We are doing a `pub use *` of everything in this module just below.